src/test/java/ca/corefacility/bioinformatics/irida/pipeline/upload/galaxy/integration/GalaxyToolDataServiceIT.java

```diff
@@ -0,0 +1,285 @@
+                    "http://s3.amazonaws.com/igv.broadinstitute.org/genomes/danRer6.genome",
+                    "http://s3.amazonaws.com/igv.broadinstitute.org/genomes/danRer7.genome",
+                    "http://s3.amazonaws.com/igv.broadinstitute.org/genomes/danRer10.genome",
+                    "https://s3.amazonaws.com/igv.broadinstitute.org/genomes/AGPv3.31.genome"
+            )
+    );
+
+
+    /**
+     * Sets up for tool data table tests.
+     */
+    @Before
+    public void setup() {
+        galaxyInstanceAdmin = localGalaxy.getGalaxyInstanceAdmin();
+        ToolDataClient toolDataClient = galaxyInstanceAdmin.getToolDataClient();
+        galaxyToolDataService = new GalaxyToolDataService(toolDataClient);
+
+    }
+
+    /**
+     * Tests getting a valid workflow input id from a workflow details.
+     * @throws GalaxyDatasetNotFoundException
+     */
+    @Test
+    public void testGetToolDataTableValid() throws GalaxyDatasetNotFoundException {
+        TabularToolDataTable toolDataTable;
+        try {
+            toolDataTable = galaxyToolDataService.getToolDataTable(VALID_TOOL_DATA_TABLE_ID);
+        } catch (Exception e) {
+            throw new GalaxyDatasetNotFoundException(e);
+        }
+        assertNotNull(toolDataTable);
+        assertEquals(VALID_TOOL_DATA_TABLE_ID, toolDataTable.getName());
+    }
+
+    /**
+     * Tests getting a valid workflow input id from a workflow details.
+     * @throws GalaxyDatasetNotFoundException
+     */
+    @Test
+    public void testGetToolDataFieldValid() throws GalaxyDatasetNotFoundException {
+        TabularToolDataTable toolDataTable;
+        try {
+            toolDataTable = galaxyToolDataService.getToolDataTable(VALID_TOOL_DATA_TABLE_ID);
+        } catch (Exception e) {
+            throw new GalaxyDatasetNotFoundException(e);
+        }
+        assertEquals("http://s3.amazonaws.com/igv.broadinstitute.org/genomes/hg38.genome",
+                galaxyToolDataService.getToolDataField(toolDataTable, VALID_TOOL_DATA_VALUE, VALID_TOOL_DATA_COLUMN));
+    }
+
+    /**
+     * Tests getting a valid workflow input id from a workflow details.
+     * @throws GalaxyDatasetNotFoundException
+     */
+    @Test
+    public void testGetToolDataColumnValid() throws GalaxyDatasetNotFoundException {
+        TabularToolDataTable toolDataTable;
+        try {
+            toolDataTable = galaxyToolDataService.getToolDataTable(VALID_TOOL_DATA_TABLE_ID);
+        } catch (Exception e) {
+            throw new GalaxyDatasetNotFoundException(e);
+        }
+        assertEquals(VALID_TOOL_DATA_COLUMN_FIELDS,
+                galaxyToolDataService.getToolDataColumn(toolDataTable, VALID_TOOL_DATA_COLUMN));
+    }
+
+    /**
+     * Tests failing to find a valid workflow input id from a workflow details.
+     * @throws GalaxyDatasetNotFoundException
+     */
+    @Test(expected=GalaxyDatasetNotFoundException.class)
```

Comment:
You could just set the expected exception to be `ClientHandlerException` and remove the `try ... catch` block.

Reply:
Thanks.